src/js/legend.js

```diff
@@ -0,0 +1,228 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
```

Comment:
After more discussion in the IRC channel - https://botbot.me/freenode/fluid-tech/2015-09-16/?tz=America/Havana, The decision is it's fine to use d3 for rendering as long as it is a reasonable infusion viewComponent: has a container, has model binding and event bind, advertise useful selectors in its DOM binder.

One improvement for the future is to make it template based so the raw materials that it generates the markup could be exposed - https://issues.fluidproject.org/browse/FLOE-419